streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -848,7 +849,7 @@ public void statelessTopologyShouldNotCreateStateDirectory() throws Exception {
         final String outputTopic = testName.getMethodName() + "-output";
         final Topology topology = new Topology();
         topology.addSource("source", Serdes.String().deserializer(), Serdes.String().deserializer(), inputTopic)
-                .addProcessor("process", () -> new AbstractProcessor<String, String>() {
+                .addProcessor("process", (ProcessorSupplier<String, String>) () -> new AbstractProcessor<String, String>() {
```

Comment:
The change in the public API has this annoying side effect that the type of the lambda cannot inferred any longer (not sure why...) -- however, this might be reason enough to not change the public API as I did above.

Let me know what you think (I though I just to the PR anyway to get a discussion going).

Reply:
I think it was never inferred before, it was actually the type `ProcessorSupplier<Object, Object>` (just my theory).

I think if old code will fail to compile, we should not change the public API. Note, the existing methods would be deprecated as part of implementing KIP-478 anyway, so maybe we should just leave the public APIs alone.